### ISSUE TITLE
Type instantiation is excessively deep and possibly infinite

### ISSUE THREAD

**Issue by christianlmc** [victorgarciaesgi/nuxt-typed-router#141]
**Describe the bug**
Installed `nuxt-typed-router` on a clean nuxt project.

Running `npx run typecheck` gave me the following error
```bash
nuxt.config.ts:4:12 - error TS2589: Type instantiation is excessively deep and possibly infinite.

4   modules: [
             ~
5     'nuxt-typed-router',
  ~~~~~~~~~~~~~~~~~~~~~~~~
6   ]
  ~~~
```
**Expected behavior**
No typescript errors

**Screenshots**
![image](https://github.com/victorgarciaesgi/nuxt-typed-router/assets/24314863/810bccc3-8f36-4db9-8f5f-0bb9a164e4db)
![image](https://github.com/victorgarciaesgi/nuxt-typed-router/assets/24314863/1b8e9ae9-997c-447d-8c10-a68daee33549)

**Environnement infos**

```bash
Working directory: /home/christian/vue/nuxt-router-bug                                                        2:37:17 PM
Nuxt project info:                                                                                            2:37:17 PM

------------------------------
- Operating System: Linux
- Node Version:     v20.11.0
- Nuxt Version:     3.10.3
- CLI Version:      3.10.1
- Nitro Version:    2.8.1
- Package Manager:  npm@10.2.4
- Builder:          -
- User Config:      devtools, modules
- Runtime Modules:  nuxt-typed-router@3.6.0
- Build Modules:    -
------------------------------
```

**Your `pages` folder structure**

Not applicable

**Your nuxt.config.ts**
```ts
// https://nuxt.com/docs/api/configuration/nuxt-config
export default defineNuxtConfig({
  devtools: { enabled: true },
  modules: [
    'nuxt-typed-router',
  ]
})
```

**Comment by victorgarciaesgi** [comment]
I'm reproducing it and don't know why it happens, looking at it thanks!

**Comment by victorgarciaesgi** [comment]
Have you reproduced it with any other modules?

**Comment by victorgarciaesgi** [comment]
After testing it happens only with Nuxt `3.10.3`, the bug isn't here with `3.10.2`. So likely a Nuxt bug

**Comment by victorgarciaesgi** [comment]
Also published a dep fix on my side on `3.6.1`

**Comment by christianlmc** [comment]
> Also published a dep fix on my side on `3.6.1`

Updated to `3.6.1` and the issue is gone, thank you so much!
